src/ucp/wireup/wireup_cm.c

```diff
@@ -963,6 +963,20 @@ ucs_status_t ucp_ep_client_cm_create_uct_ep(ucp_ep_h ucp_ep)
     cm_lane_params.disconnect_cb      = ucp_cm_disconnect_cb;
     cm_lane_params.cm                 = worker->cms[cm_idx].cm;
 
+    if (wireup_ep->cm_local_sockaddr.ss_family != 0) {
+        /* user specifed local address */
+        status = ucs_sockaddr_sizeof((const struct sockaddr*)&wireup_ep->cm_local_sockaddr,
+                                     &sockaddr_size);
+        if (status != UCS_OK) {
+            return status;
+        }
+
+        local_addr.addrlen            = sockaddr_size;
+        local_addr.addr               = (struct sockaddr*)&wireup_ep->cm_local_sockaddr;
+        cm_lane_params.field_mask    |= UCT_EP_PARAM_FIELD_LOCAL_ADDR;
+        cm_lane_params.local_sockaddr = &local_addr;
```

Comment:
why not just `cm_lane_params.local_sockaddr = &wireup_ep->cm_local_sockaddr;`

Reply:
The two fields have different types:
cm_lane_params.local_sockaddr type is "ucs_sock_addr_t" as same as existing cm_lane_params.sockaddr for server.
wireup_ep->cm_local_sockaddr type is sockaddr_storage as same as existing wireup_ep->cm_remote_sockaddr for server.

Reply:
ok, we need to save it for the possible CM fallback flow

Reply:
ok